pkg/replication/replication.go

```diff
@@ -378,6 +407,8 @@ func (r *replication) ensureHealthy(
 		select {
 		case <-quitCh:
 			return
+		case <-r.replicationCancelledCh:
```

Comment:
you're right that it's a different quitCh, I was mistaken. however my suggestion doesn't change because `updateOne` closes that channel when there is a timeout or if _its_ quit channel is closed, which is done by `Enact()` when `r.quitCh` is closed